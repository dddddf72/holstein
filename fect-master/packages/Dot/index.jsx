import { toRefs } from 'vue'
import { createNameSpace, validator, theme } from '../utils'
const { normalTypes } = theme
const [createComponent] = createNameSpace('Dot')
import './dot.less'

export default createComponent({
  props: {
    type: {
      type: String,
      validator: validator.enums(normalTypes),
      default: 'default',
    },
  },
  setup(props, { attrs, slots, emit }) {
    const { type } = toRefs(props)
    const classAttr = !!attrs?.class

    return () => (
      <>
        <div
          {...attrs}
          className={`fect-dot-wrapper ${classAttr ? attrs.class : ''}`}
        >
          <span className={`fect-dot ${type.value}`}></span>
          <span className={'fect-dot-ctx'}>{slots.default?.()}</span>
        </div>
      </>
    )
  },
})
