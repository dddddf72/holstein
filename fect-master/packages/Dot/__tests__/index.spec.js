import Dot from '../index'
import { mount } from '@vue/test-utils'
describe('Dot', () => {
  it('should be render as element', () => {
    const dot = mount(Dot)
    expect(dot.html()).toMatchSnapshot()
  })

  it('should be support types', () => {
    const wrapper = mount({
      setup() {
        return { typeList: ['default', 'success', 'warning', 'error'] }
      },
      render(h) {
        return (
          <>
            {this.typeList.map((item) => (
              <Dot type={item} />
            ))}
          </>
        )
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should be support slots', () => {
    const wrapper = mount({
      render(h) {
        return (
          <>
            <Dot type="success">
              <span>success</span>
            </Dot>
          </>
        )
      },
    })

    const el = wrapper
      .find('.fect-dot-ctx')
      .find('span')
      .text()
    expect(el).toBe('success')
    expect(wrapper.html()).toMatchSnapshot()
  })
})
