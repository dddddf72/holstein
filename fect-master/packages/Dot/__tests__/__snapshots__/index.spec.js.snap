// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dot should be render as element 1`] = `<div class="fect-dot-wrapper "><span class="fect-dot default"></span><span class="fect-dot-ctx"><!----></span></div>`;

exports[`Dot should be support slots 1`] = `<div class="fect-dot-wrapper "><span class="fect-dot success"></span><span class="fect-dot-ctx"><span>success</span></span></div>`;

exports[`Dot should be support types 1`] = `
<div class="fect-dot-wrapper "><span class="fect-dot default"></span><span class="fect-dot-ctx"><!----></span></div>
<div class="fect-dot-wrapper "><span class="fect-dot success"></span><span class="fect-dot-ctx"><!----></span></div>
<div class="fect-dot-wrapper "><span class="fect-dot warning"></span><span class="fect-dot-ctx"><!----></span></div>
<div class="fect-dot-wrapper "><span class="fect-dot error"></span><span class="fect-dot-ctx"><!----></span></div>
`;
